src/task/serverTaskLinkProvider.ts

```diff
@@ -140,13 +139,17 @@ export class serverTaskLinkProvider
 	 * @param {Array<vscode.TerminalLink>} links the array of links (pass-by-reference) that are added to.
 	 */
 
-	private _findFullLinkRegex(
+	private async _findFullLinkRegex(
 		input: string,
 		links: Array<vscode.TerminalLink>,
-		host: number
-	): void {
+		connection: Connection
+	): Promise<void> {
+		const hostUri = connection.constructLocalUri(connection.httpPort);
+		const extHostUri = await connection.resolveExternalHTTPUri();
+		const extHostStr = `${extHostUri.scheme}://${extHostUri.authority}`;
+
 		const fullLinkRegex = new RegExp(
-			`\\b\\w{2,20}:\\/\\/(?:localhost|${host}|:\\d{2,5})[\\w\\-.~:/?#[\\]@!$&()*+,;=]*`,
+			`(?:${extHostStr})[\\w\\-.~:/?#[\\]@!$&()*+,;=]*`,

```

Comment:
Does this uri need any escaping before it is put into a regex? For example will it contain special regex chars like `.*+` etc